File: ooad/shopping-app/src/com/techlabs/shop/Customer.java
package com.techlabs.shop;

import java.util.*;

public class Customer {
	private int id;
	private String name;
	private ArrayList<Order> orders;
	
	public Customer(int id, String name)
	{
		this.id = id;
		this.name = name;
		orders = new ArrayList<Order>();
	}
	
	public String getName()
	{
		return this.name;
	}
	
	public boolean addOrder(Order order)
	{
		return orders.add(order);
	}
	
	public List<Order> getOrders()
	{
		return (List<Order>)orders;
	}
}

File: typescript-app/customer.ts
export class Customer {
	constructor(private _id, private _name) { }

	get ID() {
		return this._id;
	}

	get Name() {
		return this._name;
	}

	set Name(name) {
		this._name = name;
	}
}

export class Address {
	constructor(private _address) { }

	get Address() {
		return this._address;
	}
}